mesonbuild/backend/ninjabackend.py

```diff
@@ -1808,7 +1808,7 @@ def generate_genlist_for_target(self, genlist, target, outfile):
                 elem.add_item('DEPFILE', depfile)
             if len(extra_dependencies) > 0:
                 elem.add_dep(extra_dependencies)
-            elem.add_item('DESC', 'Generating $out')
+            elem.add_item('DESC', 'Generating \'%s\'.' % sole_output)
```

Comment:
Ninja is supposed to automatically replace [`$out` with the output name](https://ninja-build.org/manual.html#_rules). Why isn't it working?

Reply:
I played around with it a bit more. So the real reason is the `quoter` variable when the ninja file is written. I worked around it by checking whether the field name is `DESC` and not quote at all. The result is I think what it was supposed to look like:

```
build yasm_app@exe/yasm.asm.o: CUSTOM_COMMAND ../src/yasm.asm
 DESC = Generating $out
 COMMAND = /usr/bin/yasm ../src/yasm.asm -o yasm_app@exe/yasm.asm.o -g dwarf2 -f elf64
```
However the final result when building is now that:
```
[3/12] Generating
```
So it looks like the `$out` variable is not evaluated at all - or is simply empty. Perhaps this variable works only for ninja `rule` lines and not `build` lines? Maybe thats also the reason in the `COMMAND` section the output is repeated as well and does not use `$out` which would have been an option as well?

Reply:
Good point, could you find other places where this mistake is being and fix those too? Also, please prefer `'Generating {!r}.format(sole_output)` over the `%` syntax.